Depend on orocos_kdl_vendor

jacobperron opened the pull request:
* Change package.xml dependency to the vendor package
* Add missing include directories for orocos_kdl
* Install tf2_kdl as a header-only target

Connects to https://github.com/ros2/ros2/pull/1208

jacobperron committed:
Depend on orocos_kdl_vendor

* Change package.xml dependency to the vendor package
* Add missing include directories for orocos_kdl
  Upstream should probably be updated to use modern cmake targets.

Signed-off-by: Jacob Perron <jacob@openrobotics.org>

sloretz commented on tf2_geometry_msgs/CMakeLists.txt:
```diff
@@ -20,7 +20,8 @@ ament_python_install_package(${PROJECT_NAME}
   "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"
-  "$<INSTALL_INTERFACE:include>")
+  "$<INSTALL_INTERFACE:include>"
+  ${orocos_kdl_INCLUDE_DIRS})
```
I would hope linking to the `orocos-kdl` target would be enough to get the include directories, so this line could be removed

jacobperron replied:
It is not enough, unfortunately. Upstream does not appear to be exporting the dependency on Eigen:

```
--- stderr: tf2_kdl                                                            
In file included from /home/jacob/ws/ros/orocos_upstream/install/orocos_kdl_vendor/include/kdl/jntarray.hpp:26,
                 from /home/jacob/ws/ros/orocos_upstream/install/orocos_kdl_vendor/include/kdl/frames_io.hpp:81,
                 from /home/jacob/ws/ros/orocos_upstream/src/ros2/geometry2/tf2_kdl/test/test_tf2_kdl.cpp:42:
/home/jacob/ws/ros/orocos_upstream/install/orocos_kdl_vendor/include/kdl/jacobian.hpp:26:10: fatal error: Eigen/Core: No such file or directory
   26 | #include <Eigen/Core>
      |          ^~~~~~~~~~~~
compilation terminated.
```

 I presume any fix upstream is unlikely to make it into Jammy though.

jacobperron committed:
Disable mem-access warnings on aarch64

Same as https://github.com/ros2/geometry2/pull/506, but in a different file
where Eigen headers are being included.

Signed-off-by: Jacob Perron <jacob@openrobotics.org>

jacobperron commented:
I had to add [8d79ded](https://github.com/ros2/geometry2/pull/473/commits/8d79ded8752a23afbef1d5cf0e14724b36fd0249) to quiet compiler warnings coming from Eigen headers.

jacobperron commented:
@ros-pull-request-builder retest this please
